biosys/apps/main/models.py

```diff
@@ -180,458 +322,185 @@ class Observation(AbstractObservationRecord):
+
     geometry = models.GeometryField(srid=MODEL_SRID, spatial_index=True, null=True, blank=True, editable=True,
-                                    verbose_name="Extent Geometry", help_text="")
-    attributes = JSONField(null=True, blank=True)
-    attributes_schema = JSONField(null=True, blank=True)
+                                    verbose_name="Extent",
+                                    help_text="The boundary of your project (not required). "
+                                              "Can also be calculated from the extents of the project sites")
+    site_data_package = JSONField(null=True, blank=True,
+                                  verbose_name='Site attributes schema',
+                                  help_text='Define here the attributes that all your sites will share. '
+                                            'This allows validation when importing sites.')
+
+    custodians = models.ManyToManyField(Group, blank=True, help_text="Add group of users that have write access"
+                                                                     " to the data of this project.")
+
+    def is_custodian(self, user):
+        # TODO: should be done in one query
+        for group in self.custodians.all():
+            if belongs_to(user, group.name):
+                return True
+        return False
+
+    # API permissions
+    @staticmethod
+    def has_read_permission(request):
+        return True
+
+    def has_object_read_permission(self, request):
+        return True
+
+    @staticmethod
+    def has_create_permission(request):
+        return is_admin(request.user)
+
+    @staticmethod
+    def has_update_permission(request):
+        """
+        The update is managed at the object level (see below)
+        :param request:
+        :return:
+        """
+        return True
+
+    def has_object_update_permission(self, request):
+        return is_admin(request.user) or self.is_custodian(request.user)
+
+    @staticmethod
+    def has_destroy_permission(request):
+        return False
+
+    def has_object_destroy_permission(self, request):
+        return False
 
     class Meta:
         pass
 
     def __str__(self):
-        return self.__unicode__()
-
-    def __unicode__(self):
         return self.title
 
-    def save(self, *args, **kwargs):
-        if self.extent_lat_min is not None and self.extent_long_min is not None and self.extent_lat_max is not None and \
-                        self.extent_long_max is not None:
-            self.geometry = Polygon.from_bbox((self.extent_long_min, self.extent_lat_min,
-                                               self.extent_long_max, self.extent_lat_max))
-
-        super(Project, self).save(*args, **kwargs)
-
-
-def _calculate_site_ID():  # @NoSelf
-    if Site.objects.count() == 0:
-        return DEFAULT_SITE_ID
-    else:
-        return Site.objects.aggregate(Max('site_ID'))['site_ID__max'] + 1
-
 
+@python_2_unicode_compatible
```

Comment:
Yep currently the code py2/py3 compatible. When I'm sure that OIM can run python3 I will probably remove all the py2 stuff (like the `six` module I use from time to time)